### PR TITLE
chore: support slack bot dependency injection

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: 

### Description:

Allow us to extend the slack bot class.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/clients/Slack/Slackbot.ts [hunk 1]
```diff
@@ -63,8 +63,6 @@ export class SlackBot {
         this.analytics = analytics;
         this.slackAuthenticationModel = slackAuthenticationModel;
         this.unfurlService = unfurlService;
-
-        void this.start();
```
Comment (ZeRego) [inline packages/backend/src/clients/Slack/Slackbot.ts:67]: Bad practice to run an async function in the constructor. We now call the `start` in App.ts.
This also allows this class methods to be overridden in classes that extend this one.

File: packages/backend/src/App.ts [hunk 2]
```diff
@@ -181,7 +198,7 @@ export default class App {
 
         const expressApp = await this.initExpress();
         this.initSentry(expressApp);
-        this.initSlack();
+        await this.initSlack();
```
Comment (IrakliJani) [inline packages/backend/src/App.ts:201]: do we need to await for this?

**Review by IrakliJani — APPROVED** [review]
one Q

File: packages/backend/src/App.ts (hunk 2 above)
Comment (ZeRego) [reply packages/backend/src/App.ts:201]: There are 3 options:
`void` - the promise doesn't block the thread but can fail silently and trigger an "unhandled promise error"
`await` - the promise blocks the thread but the error is handled
`.catch(...)` - the promise doesn't block the thread and the error is handled.

I tend to await all promises and find `void` to be the worst option of all.

**Pull request merged**